src/controller/gameController.ts

```diff
@@ -1,47 +1,85 @@
+        throw new AppError('Player not registered or currently in game.', 400);
+    }
 
     // 3) Invite players to the game
-    convertedSocket.forEach(socket => {
-        socket.emit('gameInvite');
-    });
-
-    return res.status(200).send();
+    LocalStorage.getInstance().pendingGameinvites.push(game);
+    for (const player of game.players) {
+        player.socket?.emit('gameInvite', game.toInventationJSON());
+    }
+    return res.status(200).send(game.toInventationJSON());
 });
 
-/**
- * Converts an array of objects that have the property username and socketId to an array of user objects,
- * also check whether the properties are set and throws an error if not
- * @param players array ob objects that have the property username and socketId
- * @returns Array of User Objects
- */
-const convertToSocket = (players:{username?:string, socketId?:string}[]):NopeSocket[] => {
-    const availableSockets = LocalStorage.getInstance().freeSockets.slice();
-    return players.map(player => {
-        const playerSocketIndex = availableSockets.findIndex(socket => socket.data.user?.username === player.username);
-        if ( availableSockets[playerSocketIndex] === undefined) {
-            throw new AppError('Player not registered or in game.', 400);
-        }
-        return availableSockets.splice(playerSocketIndex, 1)[0];
-    });
-};
+export function inviteResponse(socket: NopeSocket, data: InviteResponse) {
+    // check if the invite syntax is valid
+    if (data.accept === undefined || data.inviteId === undefined || data.type === undefined) {
+        throw new AppError('Invalid invite response. The Parameter accept, inviteId and type are required.', 400);
+    }
+
+    // compute type of invite
+    switch (data.type) {
+    case InviteType.GAME:
+        return gameInviteResponse(socket, data);
+    case InviteType.TOURNAMENT:
+        throw new Error('Not implemented');
+    default:
+        throw new AppError('Invalid invite type', 400);
+    }
+}
+
+function gameInviteResponse(socket : NopeSocket, data: InviteResponse) {
+    assert(data.type === InviteType.GAME, '<data.type> must be InviteType.GAME');
+    // check if the invite exists
+    const inviteGame = LocalStorage.getInstance().pendingGameinvites.find((game) => game.id === data.inviteId);
+    if (inviteGame === undefined) {
+        throw new AppError('Invite game not found', 400);
+    }
+    // check if socket was invited
+    const player = inviteGame.players.getItems().find((player) => player.socket === socket);
+    if (player === undefined) {
+        throw new AppError('You are not in the game', 400);
+    }
+    // set accept status
+    player.accepted = data.accept;
+    // check for game start
+    if (inviteGame.players.getItems().every((player) => player.accepted)) {
+        inviteGame.startGame();
+        return 'game starting';
+    }
+    return 'waiting for players';

```

Comment:
Sollten wir die Statusmeldungen wie `waiting for players` und `game starting` nicht irgendwie vordefinieren. Es wirkt etwas Hardgecoded im Code.